src/components/ContactsList/ContactsList.jsx

```diff
@@ -25,30 +23,44 @@ const ContactsList = props => {
     return <ContactsEmptyList />;
   }
   const sortedContacts = [...props.contacts].sort(sortGivenNameFirst);
-  const rows = [];
   let lastLetter = null;
-  sortedContacts.forEach(contact => {
+  const categorizedContacts = sortedContacts.reduce((acc, contact) => {
     const name = getDisplayedName(contact);
-    if (name[0] !== lastLetter) {
-      rows.push(<ContactHeaderRow header={name[0] || "EMPTY"} />);
+    const header = name[0] || "EMPTY";
+    if (header !== lastLetter) {
```

Comment:
We could do something like `if (acc[header] === undefined)` here — this would avoid the use of the `lastLetter` variable that sort of hangs out in the global scope, and it would make the function work on unsorted arrays too.

Reply:
Pas faux